Feature/ptcloud proc

JmcRobbie opened the pull request.

naverill commented on nova_rover_demos/pointCloudProcessing/ptcloudgen.py:
```diff
@@ -0,0 +1,131 @@
+			xc = int(point[0]/x_res)
+			yc = int(point[1]/y_res)
+			self.occupancy[xc][yc] = self.occupancy[xc][yc] + point[2]
+			if self.occupancy[xc][yc]>10.0:
```
Hmm I would set the max and min bounds of the point cloud to be optional parameters in the function to make it easier to configure later

JmcRobbie replied:
great idea, ill make that change as well.

naverill commented on nova_rover_demos/pointCloudProcessing/ptcloudgen.py:
```diff
@@ -0,0 +1,131 @@
+from matplotlib import pyplot
+from mpl_toolkits.mplot3d import Axes3D
+
+class point_cloud_gen:
```
Bruh what is this PEP8

JmcRobbie replied:
yeah - what is Pep8?

naverill reviewed (approved):
Code looks good and I'm not getting any errors when running it but boi does this need a good refactor. Otherwise all good to go.

naverill commented on nova_rover_demos/pointCloudProcessing/ptcloudgen.py:
```diff
@@ -0,0 +1,131 @@
+
+	def __init__ (self,size, features,noise = 1,N = 10000):
+		'''
+		size: the bounds on the range of the point cloud values. Is a list of n tuples where n is the number of dimensions
```
Also probably best to indicate the scale (m/cm) of the upper and lower bound in size

JmcRobbie replied:
good point, ill make a change.

JmcRobbie committed:
Point Cloud Processing implemented

Pushing changes for collaborators

stashing changes

implemented occupancy grid calculation function

point cloud class identifying obstacles from point cloud using height

Removed redundant helper class

Fixed some bugs and added a brief jupyter notebook demo

Actioned review comments refactored and implemented np constructor

PEP8 refactor

Tidying up some errors and cache files

Fixing up one last typo - I hope

asdf
